assets/js/components/text-toolbar-button/style.scss

```diff
@@ -7,3 +7,7 @@
 		color: $white;
 	}
 }
+.block-editor-block-toolbar__slot {
+	// prevents text toolbar items shrinking to avoid other buttons overlapping.
+	flex-shrink: 0;
+}
```

Comment:
This will leak to GB styles, can we limit it to our blocks?

Reply:
Nein it was a gutenberg style. Gutenberg only used icon buttons here, but we're inserting text buttons which is why the issue happens. This style doesn't impact the display of other buttons it seems though.